### PR TITLE
[Ath 1952] - Update components

### PR DESCRIPTION
**Pull request by EoinMcD**
# Description
- Home page has a central header, a register and login button over a mountain background
- Signup page has been styled to match login page, with a centred form, in a Sandybrown box with a white background
- The following components have been updated to use typescript
  - Navbar
  - Signup form
- Updated signup form to use AXIOS rather than AJAX for testing purposes
- Created a button component to use on the home page

# Testing
- React tests have been added for the new button component
- New tests have been added for the signup form

# Screenshots
Screenshot 2021-05-26 at 11 33 36

Screenshot 2021-05-26 at 12 28 07

### CODE REVIEW

File: app/assets/stylesheets/form.scss [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+.form {
+  width: 400px;
+  flex-direction: column;
+  height: fit-content;
+  background: sandybrown;
+  text-align: center;
+  box-shadow: 2px 5px #3d3c32e0;
+  padding: 50px;
+  border-radius: 50px;
+  justify-content: center;
```
Comment (meganduffy) [inline app/assets/stylesheets/form.scss]: Duplicated inherited value

File: app/javascript/components/button.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    name: string;
+    path: string;
+}
+const MyButton: React.FC<ButtonProps> = ({
+  name,
+  path
+}) => {
+  const classes = useStyles();
+  return (
+      <Button variant ="contained" className={classes.button} href={path}>
```
Comment (meganduffy) [inline app/javascript/components/button.tsx]: Unneeded space after `variant`

File: app/javascript/components/button.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+          {name}
```
Comment (meganduffy) [inline app/javascript/components/button.tsx]: Prop naming -> is this the name of the button? Maybe `buttonText` would be a more obvious prop name

File: app/javascript/components/signup_form.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  errorsList
+    ? (errorsList.map((error, key) =>
+                <li key = {key}>{error}</li>))
+    : null;
+
+  return (
+        <div>
+            <form className="form" onSubmit={SendForm}>
+            <h1>SIGN-UP</h1> <br></br>
+                {errorsList ? <ul className="signup-ul" data-testid="error-div">{listErrors}</ul> : null}
```
Comment (meganduffy) [inline app/javascript/components/signup_form.tsx]: As you're doing an assertion with errorsList twice, why not roll them into one?
`{ listErrors }` could conditionally render the ul with the errors mapped inside, or null.

File: test/react/button.test.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  );
+});
+
+test("Login button has the right link", () => {
+  const { container } = renderButton("/login/index");
+  expect(getByRole(container, "link")).toHaveProperty(
+    "href",
+    window.location.href + "login/index"
+  );
+});
```
Comment (meganduffy) [inline test/react/button.test.tsx]: Both of these tests are testing the same thing, that the path passed to the component renders with the correct link. We should only need to test that once. 
You probably should test that the text prop renders as we expect to have that base covered.

File: test/react/signup_form.test.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+      const { container } = renderForm();
+      expect(container.querySelector("input[name=password]")).toBeTruthy();
+    });
+    test("Password confirmation input is shown in the form", () => {
+      const { container } = renderForm();
+      expect(container.querySelector("input[name=confirmpassword]")).toBeTruthy();
+    });
+  });
+  describe("Unsuccesful logins or params", () => {
+    test("An error is shown after user hasnt correctly inputted info when submitting signup request", () => {
```
Comment (meganduffy) [inline test/react/signup_form.test.tsx]: `An error is shown when user submits incorrect data` would probably be a more accurate test name here.

File: test/react/signup_form.test.tsx [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    test("Password input is shown in the form", () => {
+      const { container } = renderForm();
+      expect(container.querySelector("input[name=password]")).toBeTruthy();
+    });
+    test("Password confirmation input is shown in the form", () => {
+      const { container } = renderForm();
+      expect(container.querySelector("input[name=confirmpassword]")).toBeTruthy();
+    });
+  });
+  describe("Unsuccesful logins or params", () => {
```
Comment (meganduffy) [inline test/react/signup_form.test.tsx]: Reference to logins, this is the signup test

File: test/react/signup_form.test.tsx [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+            "Email can't be blank",
+            "Email is invalid",
+            "Password can't be blank",
+            "Password is too short (minimum is 6 characters)",
+            "Password can't be blank"]
+        }
+      };
+      fireEvent.click(getByRole(container, "button"));
+      mockAxios.mockResponse(response);
+      expect(queryByTestId(container, "error-div")).toBeTruthy();
```
Comment (meganduffy) [inline test/react/signup_form.test.tsx:89]: You could also assert that the correct error messages are shown.

**Review by meganduffy — COMMENTED** [review]
Excellent work Eoin, just a few small things.

File: app/assets/stylesheets/form.scss (hunk 1 above)
Comment (EoinMcD) [reply app/assets/stylesheets/form.scss]: Removed duplicate value

File: app/javascript/components/button.tsx (hunk 3 above)
Comment (EoinMcD) [reply app/javascript/components/button.tsx]: Changed all instances of name to buttonText

File: test/react/button.test.tsx (hunk 5 above)
Comment (EoinMcD) [reply test/react/button.test.tsx]: Fixed now

File: test/react/signup_form.test.tsx (hunk 6 above)
Comment (EoinMcD) [reply test/react/signup_form.test.tsx]: Changed

File: test/react/signup_form.test.tsx (hunk 7 above)
Comment (EoinMcD) [reply test/react/signup_form.test.tsx]: Fixed

File: app/javascript/components/signup_form.tsx (hunk 4 above)
Comment (EoinMcD) [reply app/javascript/components/signup_form.tsx]: Rolled them into one and removed the listErrors

File: app/javascript/components/button.tsx (hunk 2 above)
Comment (EoinMcD) [reply app/javascript/components/button.tsx]: Removed space

File: test/react/signup_form.test.tsx (hunk 8 above)
Comment (EoinMcD) [reply test/react/signup_form.test.tsx:89]: Added a check for the errors

File: app/javascript/components/signup_form.tsx [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+  };
+
+  return (
+    <div>
+      <form className="form" onSubmit={SendForm}>
+        <h1>SIGN-UP</h1> <br></br>
+        {errorsList
+          ? (
+          <ul className="signup-ul" data-testid="error-div">
+            {errorsList
```
Comment (meganduffy) [inline app/javascript/components/signup_form.tsx]: Do you still need to check for the errorslist here? You're already doing it on line 49

Comment (EoinMcD) [reply app/javascript/components/signup_form.tsx]: My bad, overlooked this, fixed in the latest push

**Review by meganduffy — APPROVED** [review]
Nice work, I look forward to seeing it at demo next week!

**Pull request merged**